### ISSUE TITLE
Getting error 500 for very few vnets

### ISSUE THREAD

**Issue by rasselmaliyekkal** [gamullen/FindNextCIDRRange#8]
First of all thank you so much for this solution, we were using this solution from last few months, recently I observed, this solution is not working in rare cases, im not able to figure out whats difference between working vnets and non working.

There are no access related issues, I tried created new subnet in new RG and tested, still I get this error.

![image](https://github.com/gamullen/FindNextCIDRRange/assets/92873509/67fccf59-50f1-48f3-8849-a2ab6d856258)

![image](https://github.com/gamullen/FindNextCIDRRange/assets/92873509/4e1611df-3e80-47ae-95c7-9fcb1f6eb7b1)

![image](https://github.com/gamullen/FindNextCIDRRange/assets/92873509/f6b79a96-f84e-46de-994e-8bccbef95b41)

**Comment by rasselmaliyekkal** [comment]
When I deleted the subnet, it's working. Strange!!! In order not to miss any updates to the code, i pushed recent changes to functionapp. When i again created same subnet, its not working.. Kindly help.

![image](https://github.com/gamullen/FindNextCIDRRange/assets/92873509/ec5e67d7-a436-4c5c-825f-3b532449ceaa)

**Comment by rasselmaliyekkal** [comment]
@gamullen - What I observed, this solution is working for all existing created vnets, issue is only with newly created vnets even if it having exact same configuration with working vents. can you please test and confirm.

**Comment by rasselmaliyekkal** [comment]
@craigthackerx @AUsfy  @gamullen  - Kindly help to validate this is not a error that im alone facing, to rule out mistakes in the way i deployed the solution.

All new azure Vnets im facing the issue but not with old one

**Comment by AUsfy** [comment]
I'll check and let you know

**Comment by AUsfy** [comment]
After investigation I discovered that a field in subnets data retrieved from Azure is unexpectedly null. I think it may be related to a recent change of behavior in Azure api side which could explain why the solution is working for old vnets but not for new ones.
 I made a fix that uses the right field. @rasselmaliyekkal could you test if it's working for you ?

**Comment by rasselmaliyekkal** [comment]
@AUsfy Thank you so much, issue is fixed. I think you can proceed with merging this change to main. [ found another issue ]

**Comment by rasselmaliyekkal** [comment]
@AUsfy - Sorry, another issue occurred with this change, now it's not able to identify the existing subnet ranges.  Getting an overlapping proposed CIDR as result.

**Comment by AUsfy** [comment]
@rasselmaliyekkal  To understand well the new problem, is it only for old existing vnets ?

**Comment by rasselmaliyekkal** [comment]
@AUsfy well i haven't verified proposed cidr for new Vnets are overlapping or not.

**Comment by AUsfy** [comment]
I've tested in my side and it's working for new vnets, unfortunately I don't have old existing vnets in my environment to check.
I've made another change to the fix to support all cases. Could you check it please ?

**Comment by rasselmaliyekkal** [comment]
![image](https://github.com/gamullen/FindNextCIDRRange/assets/92873509/4f830a9d-77a6-4bb6-9c16-9916a537d345)
![image](https://github.com/gamullen/FindNextCIDRRange/assets/92873509/bfe9907c-fe23-40f4-b768-e434e8a83ab1)

@AUsfy - Still it is providing subnet range which is overlapping with existing subnet. And no issues with new Vnet.

**Comment by AUsfy** [comment]
Thanks for the tests @rasselmaliyekkal, I just realized I missed a line in the fix in which I didn't use the correct variable. 
I just made another PR  to fix it https://github.com/gamullen/FindNextCIDRRange/pull/10 . Could you please test it ?

**Comment by rasselmaliyekkal** [comment]
@AUsfy this time it worked!!!, Thank you so much, appreciate your quick response and resolution.

**Comment by AUsfy** [comment]
Great news ! wlcm ^^

**Comment by gamullen** [comment]
Merged. Thanks for the help!

**Comment by craigthackerx** [comment]
Hello! Following Gary's retirement I have now taken over this project properly. The next release (2.0.0) modernises the code (.NET 8 isolated worker, refreshed Terraform on the Libre DevOps modules) while keeping full backwards compatibility with the existing API contract.

As part of the handover I am closing the existing backlog. If this issue still applies once 2.0.0 lands, please resubmit it and I will pick it up against the new codebase. Thank you for the report, and apologies for the housekeeping churn!
